oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/hierarchy/instance/InstanceHierarchyRelationTraffic.java

```diff
@@ -0,0 +1,165 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.core.hierarchy.instance;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.skywalking.oap.server.core.analysis.Layer;
+import org.apache.skywalking.oap.server.core.analysis.MetricsExtension;
+import org.apache.skywalking.oap.server.core.analysis.Stream;
+import org.apache.skywalking.oap.server.core.analysis.metrics.Metrics;
+import org.apache.skywalking.oap.server.core.analysis.worker.MetricsStreamProcessor;
+import org.apache.skywalking.oap.server.core.remote.grpc.proto.RemoteData;
+import org.apache.skywalking.oap.server.core.source.DefaultScopeDefine;
+import org.apache.skywalking.oap.server.core.storage.StorageID;
+import org.apache.skywalking.oap.server.core.storage.annotation.BanyanDB;
+import org.apache.skywalking.oap.server.core.storage.annotation.Column;
+import org.apache.skywalking.oap.server.core.storage.type.Convert2Entity;
+import org.apache.skywalking.oap.server.core.storage.type.Convert2Storage;
+import org.apache.skywalking.oap.server.core.storage.type.StorageBuilder;
+
+@Stream(name = InstanceHierarchyRelationTraffic.INDEX_NAME, scopeId = DefaultScopeDefine.INSTANCE_HIERARCHY_RELATION,
+    builder = InstanceHierarchyRelationTraffic.Builder.class, processor = MetricsStreamProcessor.class)
+@MetricsExtension(supportDownSampling = false, supportUpdate = false)
+@EqualsAndHashCode(of = {
+    "entityId"
+}, callSuper = true)
+public class InstanceHierarchyRelationTraffic extends Metrics {
+    public static final String INDEX_NAME = "instance_hierarchy_relation";
+    public static final String SERVICE_LAYER = "service_layer";
+    public static final String INSTANCE_ID = "instance_id";
+    public static final String RELATED_SERVICE_LAYER = "related_service_layer";
+    public static final String RELATED_INSTANCE_ID = "related_instance_id";
+
+    @Setter
+    @Getter
+    @Column(name = ENTITY_ID, length = 512)
+    @BanyanDB.SeriesID(index = 0)
+    private String entityId;
+
+    @Setter
+
+
+    @Getter
+    @Column(name = INSTANCE_ID, length = 250)
+    private String instanceId;
+
+    @Setter
+    @Getter
+    @Column(name = SERVICE_LAYER)
+    private Layer servicelayer = Layer.UNDEFINED;
+
+    @Setter
+    @Getter
+    @Column(name = RELATED_INSTANCE_ID, length = 250)
+    private String relatedInstanceId;
+
+    @Setter
+    @Getter
+    @Column(name = RELATED_SERVICE_LAYER)
+    private Layer relatedServiceLayer = Layer.UNDEFINED;
```

Comment:
Once you have ID decoder, I think you don't need layer fields, because instance ID can't cross layer, right?

Reply:
we need the service layer for query because 2 instances could belong to 1 service with 2 layers.